pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -1180,10 +1165,143 @@ private static String escapeFunction(String functionName, String args, boolean s
 
   // Static variables for parsing SQL when replaceProcessing is true.
   private enum SqlParseState {
-    IN_SQLCODE,
-    ESC_TIMEDATE,
-    ESC_FUNCTION,
-    ESC_OUTERJOIN,
-    ESC_ESCAPECHAR;
+    IN_SQLCODE {
+      char[] keyword = new char[0];
+
+      @Override
+      char[] escapeKeyword() {
+        return keyword;
+      }
+    },
+    ESC_DATE {
+      char[] keyword = new char[]{'d'};
+      String replacementKeyword = "DATE ";
+
+      @Override
+      char[] escapeKeyword() {
+        return keyword;
+      }
+
+      @Override
+      String replacementKeyword() {
+        return replacementKeyword;
```

Comment:
Can this be factored out to `enum constructor`?
That is just add `char[] keyword` and `String replacement` fields to `SqlParseState` instance and avoid duplicating the logic in overridden methods.